paho/client.go

```diff
@@ -82,6 +82,11 @@ type (
 		clientInflight *semaphore.Weighted
 		debug          Logger
 		errors         Logger
+		incomingPackets chan *packets.ControlPacket
+		outgoingPackets chan *packets.ControlPacket
+		pingrespPackets chan *packets.Pingresp
+		workerErrors chan error
```

Comment:
I don't think we need the `workerErrors` channel, if the worker has an error it should just call `c.error()` which will eventually close the `stop` channel to shut everything down.